src/core/edgeNetwork/sendEdgeNetworkRequestFactory.js

```diff
@@ -81,21 +81,25 @@ export default ({
         // 204 No Content response. That's fine.
         const response = createResponse(networkResponse.parsedBody);
         cookieTransfer.responseToCookies(response);
+        return onResponseCallbackAggregator
+          .call({
+            response
+          })
+          .then(returnValues => {
+            // This line's location is very important.
+            // As long as we received a properly structured response,
+            // we consider the response sucessful enough to call lifecycle
+            // onResponse methods. However, a structured response from the
+            // server may ALSO containing errors. Because of this, we make
+            // sure we call lifecycle onResponse methods, then later
+            // process the warnings and errors.
+            // If there are errors in the response body, an error will
+            // be thrown here which should ultimately reject the promise that
+            // was returned to the customer for the command they executed.
+            processWarningsAndErrors(response);
 
-        return onResponseCallbackAggregator.call({ response }).then(() => {
-          // This line's location is very important.
-          // As long as we received a properly structured response,
-          // we consider the response sucessful enough to call lifecycle
-          // onResponse methods. However, a structured response from the
-          // server may ALSO containing errors. Because of this, we make
-          // sure we call lifecycle onResponse methods, then later
-          // process the warnings and errors.
-          // If there are errors in the response body, an error will
-          // be thrown here which should ultimately reject the promise that
-          // was returned to the customer for the command they executed.
-          processWarningsAndErrors(response);
-          return response;
-        });
+            return assign({}, ...returnValues[0], ...returnValues[1]);
```

Comment:
This looks a lot better.

Do you mind adding a comment describing what this is all about? Something like:

```
// Merges all returned objects from all `onResponse` callbacks into
// a single object that can later be returned to the customer.
```

Reply:
yes sure. thank you